flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/FunctionCatalog.java

```diff
@@ -635,12 +635,12 @@ private void validateAndPrepareFunction(CatalogFunction function)
         // it means it uses the old type inference. We assume that they have been validated before
         // being
         // wrapped.
-        if (function instanceof InlineCatalogFunction
-                && ((InlineCatalogFunction) function).getDefinition()
-                        instanceof UserDefinedFunction) {
-
+        if (function instanceof InlineCatalogFunction) {
+            // Skip validation if the input is not instance of UserDefinedFunction.
```

Comment:
// Skip validation if the input is not instance of UserDefinedFunction.
=>
            // Skip validation if it's not a UserDefinedFunction.